components/director/internal/domain/webhook/repository.go

```diff
@@ -309,6 +311,28 @@ func (r *repository) ListByWebhookType(ctx context.Context, webhookType model.We
 	return convertToWebhooks(entities, r)
 }
 
+// ListByTypeAndLabelFilter retrieves all webhooks which have the given webhook type and matching on the given label filters
+func (r *repository) ListByTypeAndLabelFilter(ctx context.Context, webhookType model.WebhookType, filter []*labelfilter.LabelFilter) ([]*model.Webhook, error) {
+	var entities Collection
+
+	filterSubquery, args, err := label.FilterQueryGlobal(model.WebhookLabelableObject, label.IntersectSet, filter)
+	if err != nil {
+		return nil, errors.Wrap(err, "while building filter query")
+	}
+
+	var conditions repo.Conditions
+	if filterSubquery != "" {
+		conditions = append(conditions, repo.NewInConditionForSubQuery("id", filterSubquery, args))
+		conditions = append(conditions, repo.NewEqualCondition("type", webhookType))
```

Comment:
This can become a one-liner: conditions = append(conditions, repo.NewInConditionForSubQuery("id", filterSubquery, args), repo.NewEqualCondition("type", webhookType))